tests/evpn_tunnel.py

```diff
@@ -0,0 +1,1143 @@
+
+
+'''
+    def entries(self, db, table):
+        tbl =  swsscommon.Table(self, db, table)
+        return set(tbl.getKeys())
+
+    def get_default_vr_id(self, dvs):
+        db = swsscommon.DBConnector(swsscommon.ASIC_DB, dvs.redis_sock, 0)
+        table = 'ASIC_STATE:SAI_OBJECT_TYPE_VIRTUAL_ROUTER'
+        tbl =  swsscommon.Table(db, table)
+        keys = tbl.getKeys()
+        assert len(keys) == 1, "Wrong number of virtual routers found"
+
+        return keys[0]
+'''
+
+class VxlanTunnel(object):
+
+    ASIC_TUNNEL_TABLE       = "ASIC_STATE:SAI_OBJECT_TYPE_TUNNEL"
+    ASIC_TUNNEL_MAP         = "ASIC_STATE:SAI_OBJECT_TYPE_TUNNEL_MAP"
+    ASIC_TUNNEL_MAP_ENTRY   = "ASIC_STATE:SAI_OBJECT_TYPE_TUNNEL_MAP_ENTRY"
+    ASIC_TUNNEL_TERM_ENTRY  = "ASIC_STATE:SAI_OBJECT_TYPE_TUNNEL_TERM_TABLE_ENTRY"
+    ASIC_BRIDGE_PORT        = "ASIC_STATE:SAI_OBJECT_TYPE_BRIDGE_PORT"
+    ASIC_VRF_TABLE          = "ASIC_STATE:SAI_OBJECT_TYPE_VIRTUAL_ROUTER"
+    ASIC_RIF_TABLE          = "ASIC_STATE:SAI_OBJECT_TYPE_ROUTER_INTERFACE"
+    ASIC_ROUTE_ENTRY        = "ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY"
+    ASIC_NEXT_HOP           = "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP"
+    ASIC_NEXT_HOP_GRP       = "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP_GROUP"
+    ASIC_NEXT_HOP_GRP_MEMBERS = "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP_GROUP_MEMBER"
+
+    tunnel_map_ids           = set()
+    tunnel_map_entry_ids     = set()
+    tunnel_map_vrf_entry_ids = set()
+    tunnel_ids               = set()
+    tunnel_term_ids          = set()
+    bridgeport_ids           = set()
+    tunnel_map_map           = {}
+    tunnel                   = {}
+    tunnel_appdb             = {}
+    tunnel_term              = {}
+    map_entry_map            = {}
+    dip_tunnel_map           = {}
+    dip_tun_state_map        = {}
+    diptunterm_map           = {}
+    bridgeport_map           = {}
+    vlan_id_map              = {}
+    vlan_member_map          = {}
+    l2mcgroup_member_map     = {}
+    l2mcgroup_map            = {}  
+    vr_map                   = {}
+    vnet_vr_ids              = set()
+    nh_ids                   = {}
+    nh_grp_id                = set()
+    nh_grp_member_id         = set()
+    route_id                 = {}
+    helper                   = None
+    switch_mac               = None
+
+
+    def __init__(self):
+        self.helper = VxlanEvpnHelper()
+
+    def create_evpn_nvo(self, dvs, nvoname, tnl_name):
+        conf_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+
+        attrs = [
+            ("source_vtep", tnl_name),
+        ]
+
+        # create the VXLAN tunnel Term entry in Config DB
+        self.helper.create_entry_tbl(
+            conf_db,
+            "VXLAN_EVPN_NVO", nvoname,
+            attrs,
+        )
+
+    def remove_evpn_nvo(self, dvs, nvoname):
+        conf_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+        self.helper.delete_entry_tbl(conf_db,"VXLAN_EVPN_NVO", nvoname,)
```

Comment:
Could you please clarify what is the reason for adding a trailing comma in function callls?
In data structures it is clear why it is needed but I am not sure about functions.
Was it done by intention?
If not then please remove extra ```,``` after last argument here and in all ```delete_entry_tbl``` and ```create_entry_tbl``` functions calls in this file.